contrib/experimental/great_expectations_experimental/expectations/expect_column_values_to_be_valid_connecticut_zip.py

```diff
@@ -0,0 +1,144 @@
+import json
+from typing import Optional
+
+import zipcodes
+
+from great_expectations.core.expectation_configuration import ExpectationConfiguration
+from great_expectations.exceptions import InvalidExpectationConfigurationError
+from great_expectations.execution_engine import (
+    PandasExecutionEngine,
+    SparkDFExecutionEngine,
+    SqlAlchemyExecutionEngine,
+)
+from great_expectations.expectations.expectation import ColumnMapExpectation
+from great_expectations.expectations.metrics import (
+    ColumnMapMetricProvider,
+    column_condition_partial,
+)
+
+
+def is_valid_connecticut_zip(zip: str):
+    list_of_dicts_of_connecticut_zips = zipcodes.filter_by(state="CT")
+    list_of_connecticut_zips = [d["zip_code"] for d in list_of_dicts_of_connecticut_zips]
+    if len(zip) > 10:
+        return False
+    elif type(zip) != str:
+        return False
+    elif zip in list_of_connecticut_zips:
+        return True
+    else:
+        return False
+
+
+# This class defines a Metric to support your Expectation.
+# For most ColumnMapExpectations, the main business logic for calculation will live in this class.
+class ColumnValuesToBeValidCTZip(ColumnMapMetricProvider):
+
+    # This is the id string that will be used to reference your metric.
+    condition_metric_name = "column_values.valid_connecticut_zip"
+
+    # This method implements the core logic for the PandasExecutionEngine
+    @column_condition_partial(engine=PandasExecutionEngine)
+    def _pandas(cls, column, **kwargs):
+        return column.apply(lambda x: is_valid_connecticut_zip(x))
+
+    # This method defines the business logic for evaluating your metric when using a SqlAlchemyExecutionEngine
+    # @column_condition_partial(engine=SqlAlchemyExecutionEngine)
+    # def _sqlalchemy(cls, column, _dialect, **kwargs):
+    #     raise NotImplementedError
+
+    # This method defines the business logic for evaluating your metric when using a SparkDFExecutionEngine
+    # @column_condition_partial(engine=SparkDFExecutionEngine)
+    # def _spark(cls, column, **kwargs):
+    #     raise NotImplementedError
+
+
+# This class defines the Expectation itself
+class ExpectColumnValuesToBeValidCTZip(ColumnMapExpectation):

```

Comment:
We'll want to keep the Expectation name & file name in concordance re: abbreviation vs full names